internal/unionstore/art/art.go

```diff
@@ -392,51 +393,86 @@ func (t *ART) Size() int {
 }
 
 func (t *ART) checkpoint() arena.MemDBCheckpoint {
-	panic("unimplemented")
+	return t.allocator.vlogAllocator.Checkpoint()
 }
 
 func (t *ART) RevertNode(hdr *arena.MemdbVlogHdr) {
-	panic("unimplemented")
+	lf := t.allocator.getLeaf(hdr.NodeAddr)
```

Comment:
Better to check the validness before using it, so the possbile wrong state would not pass around.
```
if lf == nil
   panic("To revert invalid node")
```